src/o365/graph/commands/subscriptions/subscription-add.ts

```diff
@@ -0,0 +1,244 @@
+import commands from '../../commands';
+import request from '../../../../request';
+import GlobalOptions from '../../../../GlobalOptions';
+import {
+  CommandOption, CommandValidate
+} from '../../../../Command';
+import Utils from '../../../../Utils';
+import GraphCommand from '../../../base/GraphCommand';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  resource: string;
+  changeType: string;
+  notificationUrl: string;
+  expirationDateTime: string;
+  clientState: string;
+}
+
+export const DEFAULT_EXPIRATION_DELAY_IN_MINUTES_PER_RESOURCE_TYPE = {
+  // User, group, other directory resources	4230 minutes (under 3 days)
+  "users": 4230,
+  "groups": 4230,
+  // Mail	4230 minutes (under 3 days)
+  "/messages": 4230,
+  // Calendar	4230 minutes (under 3 days)
+  "/events": 4230,
+  // Contacts	4230 minutes (under 3 days)
+  "/contacts": 4230,
+  // Group conversations	4230 minutes (under 3 days)
+  "/conversations": 4230,
+  // Drive root items	4230 minutes (under 3 days)
+  "/drive/root": 4230,
+  // Security alerts	43200 minutes (under 30 days)
+  "security/alerts": 43200
+};
+export const DEFAULT_EXPIRATION_DELAY_IN_MINUTES = 4230;
+export const SAFE_MINUTES_DELTA = 1;
+
+class GraphSubscriptionAdd extends GraphCommand {
```

Comment:
Following the naming convention, this class should be named `GraphSubscriptionAddCommand`